content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2416 @@
+learn about all the details and technicalities.
+
+- [Secret History of Prometheus
+  Histograms](https://fosdem.org/2020/schedule/event/histograms/) about the
+  classic histograms and why Prometheus kept them for so long.
+- [Prometheus Histograms – Past, Present, and
+  Future](https://promcon.io/2019-munich/talks/prometheus-histograms-past-present-and-future/)
+  is the inaugural talk about the new approach that led to native histograms.
+- [Better Histograms for
+  Prometheus](https://www.youtube.com/watch?v=HG7uzON-IDM) explains why the
+  concepts work out in practice.
+- [Native Histograms in
+  Prometheus](https://promcon.io/2022-munich/talks/native-histograms-in-prometheus/)
+  presents and explains native histograms after the actual implementation.
+- [PromQL for Native
+  Histograms](https://promcon.io/2022-munich/talks/promql-for-native-histograms/)
+  explains the usage of native histograms in PromQL.
+- [Prometheus Native Histograms in
+  Production](https://www.youtube.com/watch?v=TgINvIK9SYc) provides an analysis
+  of performance and resource consumption.
+- [Using OpenTelemetry’s Exponential Histograms in
+  Prometheus](https://www.youtube.com/watch?v=W2_TpDcess8) covers the
+  interoperability with the OpenTelemetry.
+
+## Glossary
+
+- A __native histogram__ is an instance of the new complex sample type
+  representing a full histogram that this document is about. Where the context
+  is sufficiently clear, it is often just called a _histogram_ below.
+- A __classic histogram__ is an instance of the older sample type representing
+  a histogram with fixed buckets, formerly just called a _histogram_. It exists
+  as such in the exposition formats, but is broken into a number of float
+  samples upon ingestion into Prometheus.
+- __Sparse histogram__ is an older, now deprecated name for _native
+  histogram_. This name might still be found occasionally in older
+  documentation. __Sparse buckets__ remains a meaningful term for the buckets
+  of a native histogram.
+
+## Data model
+
+This section describes the data model of native histograms in general. It
+avoids implementation specifics as far as possible. This includes terminology.
+For example, a _list_ described in this section will become a _repeated
+message_ in a protobuf implementation and (most likely) a _slice_ in a Go
+implementation.
+
+### General structure
+
+Similar to a classic histogram, a native histogram has a field for the _count_
+of observations and a field for the _sum_ of observations. In addition, it
+contains the following components, which are described in detail in dedicated
+sections below:
+
+- A _schema_ to identify the method of determining the boundaries of any given
+  bucket with an index _i_.
+- A sparse representation of indexed buckets, mirrored for positive and
+  negative observations.
+- A _zero bucket_ to count observations close to zero.
+- A (possibly empty) list of _custom values_.
+- _Exemplars_.
+
+### Flavors
```

Comment:
Perhaps standard schema and non-standard schemas is another dimension worth calling out?

Reply:
fair!